lib/widgets/login.dart

```diff
@@ -235,23 +241,119 @@ class _AddAccountPageState extends State<AddAccountPage> {
   }
 }
 
-class PasswordLoginPage extends StatefulWidget {
-  const PasswordLoginPage({super.key, required this.serverSettings});
+class LoginPage extends StatefulWidget {
+  const LoginPage({super.key, required this.serverSettings});
+
+  /// A key for the page from the last [buildRoute] call.
+  static final _lastBuiltKey = GlobalKey<_LoginPageState>();
```

Comment:
Let's order all these members in the way suggested in the Flutter style guide:
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#order-other-class-members-in-a-way-that-makes-sense

I think that'll make it somewhat easier to look at this class and understand the API.

(It looks like the existing code wasn't quite in that order, and perhaps it'd be better if it were. But that matters a lot less when it's so simple; as it gains more complexity, the structure becomes more important for the reader.)

Reply:
> It looks like the existing code wasn't quite in that order

Hmm, really? Here's the class before the commit that adds to it:

```dart
class LoginPage extends StatefulWidget {
  const LoginPage({super.key, required this.serverSettings});

  final GetServerSettingsResult serverSettings;

  static Route<void> buildRoute({required GetServerSettingsResult serverSettings}) {
    return _LoginSequenceRoute(
      page: LoginPage(serverSettings: serverSettings));
  }

  @override
  State<LoginPage> createState() => _LoginPageState();
}
```

I see the following, matching the order in the style guide:

> 1. Constructors […]

> […]

> 4. Final fields that are set from the constructor.
> 5. Other static methods [i.e. that don't return the same type as the class].

> […]

> 10. Methods […]

Reply:
I see that the updated class, with web-auth things added to it, doesn't match the numbered list; I'll fix that.

Reply:
> I see the following, matching the order in the style guide:

Oh, I guess that's right. I was reading `buildRoute` as returning the class's type, but it doesn't.

I think it's likely helpful to treat it for the ordering as if it does return the class's type, though. It plays pretty much the role of a factory constructor — it's the thing that application code, at least, should always use instead of calling the actual constructor directly. (Possibly we should even make the constructor private when we have a `buildRoute` method like this; but that might be annoying in tests.)